src/main/java/uk/gov/ons/registers/cdap/service/JSONHelper.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.ons.registers.cdap.service;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import uk.gov.ons.registers.cdap.service.tablecolumns.CompanyHouseTable;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+class JSONHelper {
+    private JSONHelper(){
+        throw new AssertionError("This class is not meant to be instantiated");
+    }
+
+    /**
+     * Method that takes byte[] HashMap of results and returns a decoded JSON object with the results
+     * Following the JSON structure of Admin Data Service
+     *
+     * String referenceColumnName is used to set the ID of the JSON object from the rowkey column in dataset being used
+     */
+    static JsonElement byteMapToGenericJSON(Map<byte[], byte[]> hashMap, String referenceColumnName){
+
+        Gson gson = new Gson();
+        JsonObject jsonData = new JsonObject();
+        JsonObject jsonVariables = new JsonObject();
+
+        // Iterates thought results and casts the byte[] objects to Strings to a JSON Object
+        for (Map.Entry<byte[], byte[]> entry : hashMap.entrySet()) {
+            String keyString;
+            String valueString;
+
+            keyString = new String(entry.getKey(), StandardCharsets.UTF_8);
+            valueString = new String(entry.getValue(), StandardCharsets.UTF_8);
```

Comment:
If you create a helper function here (eg: String decodeByterArrToString(byte[]))  the reference to charset (UTF_8) will be isolated in one place.

Additionally if you remove the separation between the declaration of keyString and valueString and the initiazliation, the code will be a bit more succinct, yet readable:

```java
String key   = decodeByterArrToString(entry.getKey());
String value = decodeByterArrToString(entry.getValue());
```

(FWIW, I miss the support for tuples. The support for tuples in Apache Commons Lang was added as a 'Pair' but I don't think we should introduce a new dependency just to use that.)